terraform/lib/dependabot/terraform/file_updater.rb

```diff
@@ -131,7 +136,7 @@ def lookup_hash_architecture # rubocop:disable Metrics/AbcSize, Metrics/MethodLe
         )
 
         base_dir = dependency_files.first.directory
-        lockfile_hash_removed = content.sub(hashes_object_regex, "")
+        lockfile_hash_removed = remove_provider_h1_hashes(content, declaration_regex)
```

Comment:
before this would remove the hashes of the _first_ provider in the lockfile. now it removes the hashes from the provider that we are trying to update so it can properly deduce the hashes for each platform and correctly match which platforms are in use.

Reply:
I think typically we would have the same platforms in all providers, only reason I've seen different ones is because people manually update a dependency and forget to run `terraform providers lock -platform=darwin_amd64 -platform=darwin_arm64 -platform=linux_amd64` :) 

So as you say the example in my reproduction would not be what you want, but doesn't seem unlikely to happen

Reply:
it would be great if it could auto recover somehow, as in you could specify which platforms you actually want.
Or if it could be deduced by looking at the provider that has the most h1 hashes 🤔 
Tho I suppose typically if you depend on multiple platforms it would be something that's solved before a dependency update happens again anyway. so maybe its not really an issue.
Its unfortunate that the platform lock is so awkward for terraform (imo) 😄

Reply:
> it would be great if it could auto recover somehow, as in you could specify which platforms you actually want

We thought about adding this feature to Dependabot in https://github.com/dependabot/dependabot-core/pull/4784#issuecomment-1064313966 but decided it would make more sense to be added upstream

I tried reaching out to the Terraform Core team about this but haven't heard back. Opening an issue in https://github.com/hashicorp/terraform for visibility might help?

The idea we proposed is similar - 

> Terraform could include something like [a Terraform resource configuration file `.tfrc` that lets users specify their platforms in plain text](https://github.com/dependabot/dependabot-core/pull/4784#issuecomment-1072700971). 
It would be helpful for tools like Dependabot to keep Terraform projects up-to-date, and also for Terraform users to more plainly define and read what architectures their projects run on.

---

> Or if it could be deduced by looking at the provider that has the most h1 hashes 🤔

This sounds like a reasonable middle ground. It might lead to some confusion though since additional platforms could be added to their provider lockfiles

FYI, right now we default to using `linux_amd64` if we can't detect the platform being used.